transform/transformer.go

```diff
@@ -152,6 +152,52 @@ func (f *RequestTransformer) TransformTimeSeries(v *prompb.TimeSeries) *prompb.T
 	return v
 }
 
+// FilterLabels operates similarly to TransformTimeSeries, but instead of modifying the TimeSeries, it calls the callback with the key and value
+// This is safe to call in parallel if the name and value bytes are not modified by the callback.
+func (f *RequestTransformer) FilterLabels(v *prompb.TimeSeries, callback func(name []byte, value []byte)) {
```

Comment:
Is `FilterLabels` an accurate name here?  It looks like the callback doesn't return a bool and the labels are not mutated so I wonder if something like `WalkLabels` (like https://pkg.go.dev/path/filepath#WalkDir) might be more appropriate.